### PR TITLE
Fix lack of confirmation when quitting via Cmd-Q

### PR DESCRIPTION
**Pull request by CelticMinstrel**
This shows a confirmation dialog when either pressing Cmd-Q (I'm guessing it also works for Alt-F4) or clicking the close button on the titlebar while a game is in progress.

This is not a complete fix for the mishandling of Cmd-Q - for example, if you press it while the recruit dialog is open, there will be no confirmation and the game will quit. Similarly, no confirmation dialog has been added to the editor. However, I think in the middle of a game is probably the most important place to require a confirmation.

### CODE REVIEW

File: src/play_controller.cpp [hunk 1]
```diff
@@ -711,6 +713,16 @@ void play_controller::process_focus_keydown_event(const SDL_Event& event)
 	}
 }
 
+void play_controller::process_quit_request() {
+	if (gui_->in_game()) {
+		const int res = gui2::show_message(gui_->video(), _("Quit"),
+					_("Do you really want to quit?"), gui2::tmessage::yes_no_buttons);
```
Comment (aginor) [inline src/play_controller.cpp]: Shouldn't this be an translatable string instead of hardcoded?

**Comment by Wedge009** [comment]
I've wondered about this as well. Good idea. Thanks.

File: src/play_controller.cpp (hunk 1 above)
Comment (CelticMinstrel) [inline src/play_controller.cpp]: Sorry, what? I only see translatable strings here.

Comment (aginor) [inline src/play_controller.cpp]: Sorry, my bad. Nevermind me :)

**Comment by shikadilord** [comment]
The behavior I observe here is that the game will display the prompt only if there isn't an active dialog. If a GUI2 dialog is active (e.g. the Quit prompt itself), the game will instantly quick as before -- which does not seem bad to me. However, if a GUI1 dialog is active instead (like the aforementioned Recruit menu, the Help browser, the Preferences dialog or the default MP lobby), nothing happens at all.

(Tested on Linux using Alt+F4, the window close button, and Ctrl+C on the console.)

**Comment by CelticMinstrel** [comment]
I do think it's bad to have instant quit for every GUI2 dialog, especially GUI2 dialogs created by add-ons as part of a scenario. I'm not sure what to do about that, though. Instant quit is fine for the titlescreen or when the quit confirmation dialog is up, I think.

**Comment by shikadilord** [comment]
Hm, now choosing No in the confirmation prompt displayed when attempting to close from a running game will cause Wesnoth to quit anyway. Perhaps the event is being re-caught by one of the ever-present GUI1 buttons from the in-game themeable UI?

**Pull request closed without merging**